File: S/260-single-number-iii/main.go
package main

import "fmt"

func singleNumber(nums []int) []int {
	sum := 0
	for i := 0; i < len(nums); i++ {
		sum ^= nums[i]
	}

	flag := (-sum) & sum
	a, b := 0, 0
	for i := 0; i < len(nums); i++ {
		if flag&nums[i] == 0 {
			a ^= nums[i]
		} else {
			b ^= nums[i]
		}
	}
	return []int{a, b}
}

func main() {
	fmt.Println("a")
}

File: S/260-single-number-iii/main.rs
struct Solution {}

impl Solution {
  pub fn single_number(nums: Vec<i32>) -> Vec<i32> {
    let mut sum: i32 = 0;
    nums.iter().for_each(|&x| sum ^= x);

    let flag = (-sum) & sum;
    let mut a: i32 = 0;
    let mut b: i32 = 0;
    nums.iter().for_each(|&x| {
      if flag & x == 0 {
        a ^= x;
      } else {
        b ^= x;
      }
    });
    vec![a, b]
  }
}
